src/server/__tests__/server.unit.spec.js

```diff
@@ -10,50 +12,66 @@ jest.mock('@hapi/hapi', () => {
       info: {
         uri: 'http://localhost:5000'
       },
-      register: jest.fn()
+      register: jest.fn(),
+      realm: { modifiers: { route: {} } }
     }))
   }
   return Hapi
 })
 
-describe('Server Setup', () => {
-  let logSpy
-  let errorSpy
+afterEach(() => {
+  jest.restoreAllMocks()
+})
 
-  beforeAll(() => {
-    logSpy = jest.spyOn(console, 'log').mockImplementation(() => {})
-    errorSpy = jest.spyOn(console, 'error').mockImplementation(() => {})
-  })
+describe('server.unit', () => {
+  it('should log a message saying the server has started successfully', async () => {
+    sequelize.authenticate.mockResolvedValueOnce()
 
-  afterAll(() => {
-    logSpy.mockRestore()
-    errorSpy.mockRestore()
-  })
+    await initialiseServer()
 
-  beforeEach(() => {
-    jest.clearAllMocks()
+    expect(logger.info).toHaveBeenCalledWith(
+      'Server started at %s. Listening on %s',
+      expect.any(Date),
+      expect.any(String)
+    )
   })
 
   it('should log successful connection message when database connection is successful', async () => {
-    sequelize.authenticate.mockResolvedValue()
+    sequelize.authenticate.mockResolvedValueOnce()
 
     await initialiseServer()
 
-    expect(logSpy).toHaveBeenCalledWith(
+    expect(logger.info).toHaveBeenCalledWith(
       'Connection has been established successfully.'
     )
   })
 
   it('should log an error message when database connection fails', async () => {
     const errorMessage = 'Unable to connect to the database'
 
-    sequelize.authenticate.mockRejectedValue(new Error(errorMessage))
+    sequelize.authenticate.mockRejectedValueOnce(new Error(errorMessage))
 
     await initialiseServer()
 
-    expect(errorSpy).toHaveBeenCalledWith(
+    expect(logger.error).toHaveBeenCalledWith(
       'Unable to connect to the database:',
       expect.any(Error)
     )
   })
+
+  it('should log an error and exit on unhandledRejection', async () => {
+    sequelize.authenticate.mockResolvedValueOnce()
+    const mError = new Error('Unexpected error')
+    jest.spyOn(process, 'on').mockImplementation((event, handler) => {
+      if (event === 'unhandledRejection') {
+        handler(mError)
+      }
+    })
+    const exitSpy = jest.spyOn(process, 'exit').mockReturnValueOnce()
+
+    await initialiseServer()
+
+    expect(logger.error).toHaveBeenCalledWith(new Error('Unexpected error'))
+    expect(exitSpy).toHaveBeenCalledWith(1)
+  })
 })
```

Comment:
These are a great start, but I think you also need to test sequelize.authenticate, the server.route calls, etc. Imagine if you'd constructed server.js with TDD, so writing a failing test, then writing the code to make it pass. I think we're getting a bit beyond the scope of this PR's intent though, so I don't mind if you add in a new ticket to cover the unit testing of existing modules